Pod/Classes/Utils.m

```diff
@@ -323,76 +322,45 @@ + (BOOL)isExt:(NSString *)ext In:(__strong NSString *[])exts
-
-    ALAssetRepresentation *rep = [asset defaultRepresentation];
-    uint8_t *buffer = calloc(BufferSize, sizeof(*buffer));
-    NSUInteger offset = 0, bytesRead = 0;
-
-    do {
-        @try {
-            bytesRead = [rep getBytes:buffer fromOffset:offset length:BufferSize error:nil];
-            [handle writeData:[NSData dataWithBytesNoCopy:buffer length:bytesRead freeWhenDone:NO]];
-            offset += bytesRead;
-        } @catch (NSException *exception) {
-            free(buffer);
-            return NO;
-        }
-    } while (bytesRead > 0);
-
-    free(buffer);
-    return YES;
-}
-
-+ (BOOL)writeDataToPathWithMeta:(NSString*)filePath andAsset:(ALAsset*)asset
-{
-    @autoreleasepool {
-        ALAssetRepresentation *defaultRep = asset.defaultRepresentation;
-        CGImageRef cgimg = [defaultRep CGImageWithOptions:defaultRep.metadata];
-        UIImage *image = [UIImage imageWithCGImage:cgimg];
-        CGImageSourceRef source =  CGImageSourceCreateWithData((CFDataRef)UIImageJPEGRepresentation(image, 1.0), NULL);
-        
++ (BOOL)writeDataToPathWithMeta:(NSString*)filePath andAsset:(PHAsset *)asset {
+    PHImageRequestOptions *options = [PHImageRequestOptions new];
+    options.networkAccessAllowed = YES; // iCloud
+    options.deliveryMode = PHImageRequestOptionsDeliveryModeHighQualityFormat;
+    options.synchronous = YES;
+    
+    __block BOOL success = NO;
+    [[PHImageManager defaultManager] requestImageDataForAsset:asset options:options resultHandler:^(NSData * _Nullable imageData, NSString * _Nullable dataUTI, UIImageOrientation orientation, NSDictionary * _Nullable info) {
+//        NSError *error = nil;
         NSURL *url = [[NSURL alloc] initFileURLWithPath:filePath];
-        CFStringRef UTI = CGImageSourceGetType(source); //this is the type of image (e.g., public.jpeg)
-        CGImageDestinationRef destination = CGImageDestinationCreateWithURL((CFURLRef)url, UTI, 1, NULL);
-        if(!destination) {
-            Debug("***Could not create image destination ***");
-            return false;
-        }
-        
-        //add the image contained in the image source to the destination, overidding the old metadata with our modified metadata
-        CGImageDestinationAddImageFromSource(destination, source, 0, (CFDictionaryRef)defaultRep.metadata);
-        
-        //tell the destination to write the image data and metadata into our data object.
-        //It will return false if something goes wrong
-        BOOL success = CGImageDestinationFinalize(destination);
-        if (!success) {
-            Debug("***Could not create data from image destination ***");
-        }
-        CFRelease(destination);
-        CFRelease(source);
-        return success;
-    }
+        success = [imageData writeToURL:url atomically:true];
+    }];
+    
+    return success;
 }
 
-+ (BOOL)writeDataToPath:(NSString*)filePath andAsset:(ALAsset*)asset
++ (BOOL)writeDataToPath:(NSString*)filePath andAsset:(PHAsset *)asset
```

Comment:
what is the difference of this function from the above one?
We need to refactor these two functions